# **Protocol: a detailed guide to assess genome assembly based on long-read sequencing data**
Scenario show case for genome assembly based on long-read sequencing data using Inspector. 

Inspector package can be found [https://github.com/ChongLab/Inspector](https://github.com/ChongLab/Inspector)

---
## **Description**
Inspector is a tool for assembly evaluation with long read data. 

---

## **Dependency**
| Dependencies for Inspector:|
| --- | 
| python (tested with version 3.12.3) |
| pysam (tested with version 0.22.1) |
| statsmodels (tested with version 0.14.2) |
| minimap2 (tested with version 2.28) |
| samtools (tested with version 1.20) |

|Dependencies for Inspector error correction module:|
| --- | 
| flye (tested with version 2.9.5) |

## **Installation**

To create an environment with conda (recommended):
```bash
mamba create --name ins inspector
mamba activate ins
```
Git install after installing all the dependencies.
```bash
git clone https://github.com/ChongLab/Inspector.git
export PATH=$PWD/Inspector/:$PATH
```

---

## **Scenario 1: Reference-free diploid and haplotype-resolved genome assemblies evaluation**

### **Stage 1: Download both assembly and raw genome reads files.**

Scenario 1 assembly files:

- hap1.fa.gz
- hap2.fa.gz

```bash
cd /path/to/assembled_contig/
wget https://s3-us-west-2.amazonaws.com/human-pangenomics/submissions/53FEE631-4264-4627-8FB6-09D7364F4D3B--ASM-COMP/HG002/assemblies/hifiasm_v0.19.5/hic/HG002.hap1.fa.gz
wget https://s3-us-west-2.amazonaws.com/human-pangenomics/submissions/53FEE631-4264-4627-8FB6-09D7364F4D3B--ASM-COMP/HG002/assemblies/hifiasm_v0.19.5/hic/HG002.hap2.fa.gz
```

Scenario 1 Raw reads files:

- m64011_190830_220126.Q20.fastq.gz
- m64011_190901_095311.Q20.fastq.gz
- m64012_190920_173625.Q20.fastq.gz
- m64012_190921_234837.Q20.fastq.gz

```bash
cd /path/to/rawfastq/
wget https://storage.googleapis.com/brain-genomics/awcarroll/t2t/fastq/q20/m64011_190830_220126.Q20.fastq.gz
wget https://storage.googleapis.com/brain-genomics/awcarroll/t2t/fastq/q20/m64011_190901_095311.Q20.fastq.gz
wget https://storage.googleapis.com/brain-genomics/awcarroll/t2t/fastq/q20/m64012_190920_173625.Q20.fastq.gz
wget https://storage.googleapis.com/brain-genomics/awcarroll/t2t/fastq/q20/m64012_190921_234837.Q20.fastq.gz
```

### **Stage 2: Genome assembly evaluation using Inspector for each haplotype**

Scenario 1 settings:

- Raw reads type used for evaluation: HiFi
- Reference: no
- Error correction: yes

```bash
inspector.py -c /path/to/assembled_contig/HG002.hap1.fa \
-r /path/to/rawfastq/*.fastq.gz -o HG002_HiFi_hap1_noref/ -t8 --datatype hifi \
1> HG002_HiFi_hap1_noref.log 2>&1
inspector-correct.py -t8 -i HG002_HiFi_hap1_noref/ --datatype pacbio-hifi \
-o HG002_HiFi_hap1_noref

inspector.py -c /path/to/assembled_contig/HG002.hap2.fa \
-r /path/to/rawfastq/*.fastq.gz -o  HG002_HiFi_hap2_noref/ -t8 --datatype hifi \
1> HG002_HiFi_hap2_noref.log 2>&1
inspector-correct.py -t8 -i HG002_HiFi_hap2_noref/ --datatype pacbio-hifi \
-o HG002_HiFi_hap2_noref
```

---

## **Scenario 2: Reference-free HiFi primary genome assembly evaluation with error correction**

### **Stage 1: Download both assembly and raw genome reads files.**

Scenario 2 assembly file:

- HiFi.hifiasm-0.12.pri.fa.gz

```bash
cd /path/to/assembled_contig/
wget https://zenodo.org/records/4393631/files/HG00733.HiFi.hifiasm-0.12.pri.fa.gz
```

Scenario 2 raw reads run IDs from Sequence Read Archive:

- ERR3822935
- ERR3861382
- ERR3861383
- ERR3861384
- ERR3861385
- ERR3861386
- ERR3861387

Download SRR_Acc_List.txt (select all 7 runs) form [https://www.ncbi.nlm.nih.gov/Traces/study/?acc=ERX3831682&o=acc_s%3Aa&s=ERR3822935](https://www.ncbi.nlm.nih.gov/Traces/study/?acc=ERX3831682&o=acc_s%3Aa&s=ERR3822935)

```bash
$ cd /path/to/rawfastq/
$ while read sample; do fastq-dump  $sample ; done <SRR_Acc_List.txt
```

### **Stage 2: Genome assembly evaluation using Inspector**

Scenario 2 settings:

- Raw reads type used for evaluation: HiFi
- Reference: no
- Error correction: yes

```bash
inspector.py -c /path/to/assembled_contig/HG00733.HiFi.hifiasm-0.12.pri.fa \
-r /path/to/ rawfastq/*.fastq -o HG00733_HiFi_noref/ -t8 --datatype hifi \
1>HG00733_HiFi_noref.log 2>&1
inspector-correct.py -t8 -i HG00733_HiFi_noref/ --datatype pacbio-hifi -o HG00733_HiFi_noref/
```

---

## **Scenario 3: Reference-based ONT genome assembly evaluation with error correction**

### 

**Stage 1: Download assembly, raw genome reads files, and genome reference file**

Scenario 3 assembly filename:

- ONT.Flye-2.4.2_Helen.pri.fa.gz

```bash
cd /path/to/assembled_contig/
wget https://zenodo.org/records/4393631/files/HG00733.ONT.Flye-2.4.2_Helen.pri.fa.gz

```

Scenario 3 raw reads file names:

- HG00733_1.fastq.gz
- HG00733_2.fastq.gz
- HG00733_3.fastq.gz

```bash
cd /path/to/rawfastq/
wget https://s3-us-west-2.amazonaws.com/human-pangenomics/working/HPRC_PLUS/HG00733/raw_data/nanopore/HG00733_1.fastq.gz
wget https://s3-us-west-2.amazonaws.com/human-pangenomics/working/HPRC_PLUS/HG00733/raw_data/nanopore/HG00733_2.fastq.gz 
wget https://s3-us-west-2.amazonaws.com/human-pangenomics/working/HPRC_PLUS/HG00733/raw_data/nanopore/HG00733_3.fastq.gz
```

Scenario 3 reference FASTA file (T2T-CHM13 v2.0/hs1):

- File name: hs1.fa.gz

```bash
cd Â /path/to/reference_genome/
wget https://hgdownload.soe.ucsc.edu/goldenPath/hs1/bigZips/hs1.fa.gz
```

### **Stage 2: Genome assembly evaluation using Inspector**

Scenario 3 settings:

- Raw reads type used for evaluation: ONT
- Reference: yes
- Error correction: yes

```bash
inspector.py -c /path/to/assembled_contig/HG00733.ONT.Flye-2.4.2_Helen.pri.fa \
	-r /path/to/ rawfastq/ *.fastq.gz --ref /path/to/reference_genome/hs1.fa \
	-o HG00733_ONT_ref -t8 --datatype nanopore 1>HG00733_ONT_ref.log 2>&1
inspector-correct.py -t8 -i HG00733_ONT_ref/ --datatype nano-raw -o HG00733_ONT_ref/

```

---

## **Scenario 4: Reference-only ONT genome assembly evaluation with error correction**

### **Stage 1: Download assembly, raw genome reads files, and genome reference file**

Scenario 4 is a reference-only evaluation, where the assembled contigs and the reference genome are the same as those in Scenario dataset 3.

### **Stage 2: Genome assembly evaluation using Inspector**

Scenario 4 settings:

- Raw reads type used for evaluation: none
- Reference: yes
- Error correction: no

```bash
inspector.py -c /path/to/assembled_contig/HG00733.ONT.Flye-2.4.2_Helen.pri.fa \
	-r emptyfile --ref /path/to/reference_genome/hs1.fa -o HG00733_ONT_ref_only \
	-t8 1>HG00733_ONT_ref_only.log 2>&1

```